src/content/language/mixins.md

```diff
@@ -60,30 +61,104 @@ mixin Musical {
 }
 ```
 
-Sometimes you might want to restrict the types that can use a mixin.
-For example, the mixin might depend on being able to invoke a method
-that the mixin doesn't define.
-As the following example shows, you can restrict a mixin's use
-by using the `on` keyword to specify the required superclass:
+## Specify members a mixin can call on itself
+
+Sometimes a mixin depends on being able to invoke a method or access fields,
+but can't define those members itself (because mixins can't use constructor
+parameters to instantiate their own fields).
+
+The following sections cover different strategies for ensuring any subclass
+of a mixin defines any members the mixin's behavior depends on. 
+
+### Define abstract members in the mixin
+
+By declaring the mixin as abstract, you force any type that uses
+it to define the abstract method upon which its behavior depends. 
+
+```dart
+abstract mixin Musician {
+  void playInstrument(String instrumentName); // Abstract method.
+
+  void playPiano() {
+    playInstrument('Piano');
+  }
+  void playFlute() {
+    playInstrument('Flute');
+  }
+}
+
+class Virtuoso with Musician { 
+  void playInstrument(String instrumentName) { // Subclass must define.
+    print('Plays the $instrumentName beautifully');
+  }  
+} 
+```
+
+#### Access state in the mixin's subclass
+
+Declaring abstract memebers also allows you to access state on the subclass
+of a mixin, by calling getters which are defined as abstract on the mixin:
+
+```dart
+/// Can be applied to any type with a [name] property and provides an
+/// implementation of [hashCode] and operator `==` in terms of it.
+mixin NameIdentity {
+  String get name;
+
+  int get hashCode => name.hashCode;
+  bool operator ==(other) => other is NameIdentity && name == other.name;
+}
+
+class Person with NameIdentity {
+  final String name;
+
+  Person(this.name);
+}
+```
+
+### Implement an interface
+
+Similar to declaring the mixin abstract, putting an `implements` clause on the
+mixin while not actually implementing the interface will also ensure any member
+dependencies are defined for the mixin.
+
+```dart
+?
```

Comment:
Don't forget to fill this in. :)

Reply:
Oh! I'm sorry, I think I actually needed help with this example, not a review yet 😬 

Would it look something like... this? Not sure what would go in the mixin

```dart
class Person {
  final String _name;

  Person(this._name);

  String greet(String who) => 'Hello, $who. I am $_name.';
}

mixin Impersonator implements Person {
  // what goes in here?
}

class Impostor with Impersonator {
  String get _name => '';

  String greet(String who) => 'Hi $who. Do you know who I am?';
}
```

Reply:
Maybe something like:

```dart
abstract interface class Tuner {
  void tuneInstrument();
}

mixin Guitarist implements Tuner {
  void playSong() {
    // Should be in tune first!
    tuneInstrument();

    print('Strums guitar majestically.');
  }
}

class PunkRocker implements Tuner with Guitarist {
  void tuneInstrument() {
    print("Don't bother, being out of tune is punk rock.");
  }
}
```

?

Reply:
I don't think the class using the mixin also needs to implement Tuner, right? I wrote it like `class PunkRocker with Guitarist {}` and that seems to work fine.